src/main/flight/gps_rescue.c

```diff
@@ -540,7 +556,7 @@ static void sensorUpdate(void)
 
     if (rescueState.phase == RESCUE_LANDING) {
         // do this at sensor update rate, not the much slower GPS rate, for quick disarm
-        rescueState.sensor.accMagnitude = (float) sqrtf(sq(acc.accADC[Z]) + sq(acc.accADC[X]) + sq(acc.accADC[Y])) * acc.dev.acc_1G_rec;
+        rescueState.sensor.accMagnitude = (float) sqrtf(sq(acc.accADC[Z] - acc.dev.acc_1G) + sq(acc.accADC[X]) + sq(acc.accADC[Y])) * acc.dev.acc_1G_rec;
```

Comment:
```suggestion
        // Warning: this line assumes that the gravity vector is perfectly aligned with the attitude vector at all times
        rescueState.sensor.accMagnitude = (float) sqrtf(sq(acc.accADC[Z] - acc.dev.acc_1G) + sq(acc.accADC[X]) + sq(acc.accADC[Y])) * acc.dev.acc_1G_rec;
```
This line looks dangerous. It assumes that the quad is perfectly horizontal at all times. Since the quad should be pretty much level in the landing phase this is a somewhat good assumption. But at least there should be a comment that is warning about this assumption, imo.

Reply:
Better would just be to calculate the true non gravity acceleration that the drone is experiencing. That will be used to fuse gps and acc anyhow

Reply:
Good point.  I'll add a warning.
This works much more reliably than the previous 'assumptions' :-)
I agree that a value that provided non-gravity acceleration would be much better; it would be great if that can be provided at some point in the future.
If, during landing phase the quad is on a steeper angle, that's likely part of an awkard landing, or it is rolling over, and if that outcome in the existing code results in a greater sensitivity to disarm, this 'peculiarity' is, of itself, not so bad.